### PR TITLE
Give min requirements to VS Code integrated terminal + debugger

### PR DESCRIPTION
**Pull request by wuweiweiwu**
The debugger is not a full terminal emulator thus fails the `stream.isTTY` check. However, it still supports color. Thus by checking if the process is run through vscode through `process.env.VSCODE_PID` we can give the min color support to the debugger. (The terminal currently has level 1 support and the debugger has none)

```javascript
// what the vscode integrated terminal has right now
{ supportsColor: [Function: getSupportLevel],
  stdout: { level: 1, hasBasic: true, has256: false, has16m: false },
  stderr: { level: 1, hasBasic: true, has256: false, has16m: false } }
```

https://github.com/Microsoft/vscode/issues/3006#issuecomment-190664166

issue from Chalk:
https://github.com/chalk/chalk/issues/254

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ function supportsColor(stream) {
 		return 2;
 	}
 
+	if (env.VSCODE_PID) {
```
Comment (kevva) [inline index.js]: Doesn't the integrated terminal set the correct environment flags already? I thought it was only the debugger that didn't.

Comment (wuweiweiwu) [reply index.js]: My mistake. I'll move this check down to the if block on line 54 because the debugger doesnt have `isTTY` set

**Comment by Qix-** [comment]
Thank you :)

**Comment by wuweiweiwu** [comment]
No problem!

**Pull request merged**